test/rubygems/test_gem.rb

```diff
@@ -740,6 +740,27 @@ def test_self_latest_spec_for
     assert_equal gems['a-2'], spec
   end
 
+  def test_self_latest_spec_for_multiple_sources
+    uri = 'https://example.sample.com/'
+    source = Gem::Source.new(uri)
+    source_list = Gem::SourceList.new
+    source_list << Gem::Source.new(@uri)
+    source_list << source
+    Gem.sources.replace source_list
+
+    spec_fetcher(uri) do |fetcher|
+      fetcher.spec 'a', 1.1
+    end
+
+    gems = spec_fetcher do |fetcher|
+      fetcher.spec 'a', 1
+      fetcher.spec 'a', '3.a'
+      fetcher.spec 'a', 2
+    end
+    spec = Gem.latest_spec_for 'a'
+    assert_equal gems['a-2'], spec
```

Comment:
the problem happens when `multiple sources are used, and a gem exists in both`
but this test is not testing that, it seems that `a-2` is  only in one of the resources not both, i'm missing something?

Reply:
Am I misunderstanding? The gem is `a`, with different versions `1`, `1.1`, `2`, `3.a`. It's possible for not all versions to be in all sources. In my case, which is likely an edge case, the gem was forked to an internal hosted rubygems source, and isn't updated regularly.